### PR TITLE
Fix add/publish so we wait for the state to be loaded

### PR DESCRIPTION
**Pull request by arj03**
This fixed the problem found in https://github.com/ssbc/ssb-friends/pull/49. Namely that we don't wait for the state to be loaded, so we end up with the wrong validate. Only publish and add really need this anyway which is why they are the only ones with this check.

### CODE REVIEW

**Comment by arj03** [comment]
Must have been the random failure we have seen one time before. Works locally.

File: test/basic.js [hunk 1]
```diff
@@ -157,11 +157,57 @@ test('delete all', (t) => {
             (err, results) => {
               t.error(err, 'no err')
               t.equal(results.length, 0, 'gone')
-              sbot.close(t.end)
+              t.end()
```
Comment (staltz) [inline test/basic.js:160]: Since we're creating a new sbot, couldn't we close this previous one?

Comment (staltz) [reply test/basic.js:160]: Nevermind!

**Review by staltz — APPROVED** [review]
Nice find!

**Pull request merged**